test-case/check-reboot.sh

```diff
@@ -0,0 +1,135 @@
+delay=${OPT_VALUE_lst['d']}
+
+# write the orig & status count to the status file
+status_log=$LOG_ROOT/status.txt
+echo $loop_count >> $status_log
+count=$(head -n 1 $status_log|awk '{print $1;}')
+# because write record by desc
+current=$[ $count - $loop_count ]
+
+dlogi "Round: $current/$count"
+
+# verify-pcm-list.sh & verify-tplg-binary.sh need TPLG file
+export TPLG=$(sof-get-default-tplg.sh)
+
+declare -a verify_lst
+verify_lst=(${verify_lst[*]} "verify-firmware-presence.sh")
+verify_lst=(${verify_lst[*]} "verify-kernel-module-load-probe.sh")
+verify_lst=(${verify_lst[*]} "verify-pcm-list.sh")
+verify_lst=(${verify_lst[*]} "verify-sof-firmware-load.sh")
+verify_lst=(${verify_lst[*]} "verify-tplg-binary.sh")
+
+# because for the 1st round those information write by nohup command redirect
+# but after system reboot the nohup redirect is missing
+dlogi "Check status begin"
+for i in ${verify_lst[*]}
+do
+    dlogc "$(dirname ${BASH_SOURCE[0]})/$i"
+    $(dirname ${BASH_SOURCE[0]})/$i
+    if [ $? -ne 0 ];then
+        # last line add failed keyword
+        sed -i '$s/$/ fail/' $status_log
+        dlogi "$i: fail" 
+        exit 1
+    else
+        # last line add pass keyword
+        dlogi "$i: pass" 
+    fi
+done
+dlogi "Round $current: Status check finished"
+sed -i '$s/$/ pass/' $status_log
+[[ $loop_count -le 0 ]] && exit 0
+
+dlogi "Do the perpare for the next round bootup"
+# run the script for the next round
+next_count=$[ $loop_count - 1 ]
+
+full_cmd=$(ps -p $PPID -o args --no-header)
+# parenet process have current script name
+# like: bash -c $0 .....
+if [[ "$full_cmd" =~ "bash -c" ]]; then
+    full_cmd=${full_cmd#bash -c}
+else
+    full_cmd=$(ps -p $$ -o args --no-header)
+    full_cmd=${full_cmd#\/bin\/bash}
+fi
+
+# load script default value for the really full command
+if [ $# -eq 0 ]; then
+    full_cmd=$(echo $full_cmd|sed "s:$0:& -l $loop_count:g")
+fi
+
+# convert relative path to absolute path
+full_cmd=$(echo $full_cmd|sed "s:$0:$(realpath $0):g")
+# some load will use '~' which is $HOME, but after system reboot, in rc.local $USER is root
+# so the '~' will direct the error path
+full_cmd=$(echo $full_cmd|sed "s:~:$HOME:g")
+# now convert full current command to next round command
+full_cmd=$(echo $full_cmd|sed "s:-l $loop_count:-l $next_count:g")
+
+boot_file=/etc/rc.local
+# if miss rc.local file let sof-boot-once.sh to create it
+[[ ! -f $boot_file ]] && sudo sof-boot-once.sh
+
+# because write too many information & complex to the /etc/rc.local file, so here direct operation it
+# normally suggest to use 'sudo sof-boot-once.sh command' to write it
+
+# change the file own & add write permission
+sudo chmod u+w $boot_file
+sudo chown $UID $boot_file
+# because hijack sudo command, so "sed -i '/^exit 0/d' $boot_file" have some problem in here
+old_content="$(cat $boot_file|grep -v '^exit')"
+# write the infromation to /etc/rc.local
+# LOG_ROOT to confirm all data include sub-case write log to target folder
```

Comment:
I think what you meant is,  "LOG_ROOT to make sure all tests, including sub-cases, write log to the same target folder"